src/ooo_cpu.cc

```diff
@@ -30,6 +30,24 @@ void O3_CPU::read_from_trace()
 
         if (knob_cloudsuite) {
             if (!fread(&current_cloudsuite_instr, instr_size, 1, trace_file)) {
+	      if((SINGLE_CORE_TRACE_END_EXIT==1) && (NUM_CPUS==1))
```

Comment:
I don't think I like the way you've implemented this here. It feels very ham-handed to have this as a preprocessor define. I feel like there are multiple ways you can handle terminating on trace end: terminate when the first trace ends, when the last trace ends, or never. Could we change this to a CLI option like `--trace-end=(first|last|loop)`?

Reply:
I think it'll do until we move more into reworking how champsim is configured at run time.